src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -303,14 +296,17 @@ SymbolKind.NetModule or
                 CancellationToken cancellationToken)
             {
                 var state = ReadState();
-                var compilationWithoutGeneratedDocuments = state is WithCompilationTrackerState withCompilationState
-                    ? withCompilationState.CompilationWithoutGeneratedDocuments
+                var compilationWithoutGeneratedDocuments = state is WithCompilationTrackerState withCompilationState1
+                    ? withCompilationState1.CompilationWithoutGeneratedDocuments
                     : null;
 
+                generatorInfo = state is WithCompilationTrackerState withCompilationState2
+                    ? withCompilationState2.GeneratorInfo
+                    : CompilationTrackerGeneratorInfo.Empty;
```

Comment:
My initial thought was "why do we have two different states?" Is this worth either just using a strongly typed local and then the ?., or more fun use a lambda/deconstruction?

Reply:
Or: is it just easier to special case the no-compilation scenario entirely? Then everything else here is with compilation branch?

Reply:
yup.  i'll go with the latter.